### PR TITLE
Add pacemaker qdevice management

### PR DESCRIPTION
**Pull request by OliPou**
This modification add the qdevice management in a pacemaker cluster

### CODE REVIEW

**Comment by OndrejHome** [comment]
Hello @OliPou,

Thank you also for this PR. I will take a look and run few tests with it on my test systems first and later let you know what modifications to code are needed.

From a quick glance at code I see so far:
- `author: "Olivier Pouilly (@OndrejHome)"` -> `author: "Olivier Pouilly (@OliPou)"` 
- lines 132-138 contains just commented out code

Once I will get through testing this out I will provide you with more feedback. Thank you for your patience.

Ondrej (온드레이)

File: library/pcs_quorum_qdevice [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    qdevice: qdevice-name
+    algorithm: ffsplit
+  run_once: True
+
+- name: Setup qdevice
+  pcs_quorum_qdevice:
+    qdevice: qdevice-name
+    algorithm: lms
+    allowed_qdevice_changes: update
+  run_once: True
```
Comment (OndrejHome) [inline library/pcs_quorum_qdevice:84]: (low) would it be possible to make descriptions of tasks a bit more descriptive, for example:
- Setup qdevice with default algorithm (ffsplit)
- Setup qdevice with lms algorithm
- Delete qdevice
- Setup or modify qdevice to use lms algorith

Note I have also removed one duplicate 'Setup qdevice' with ffsplit.

**Review by OndrejHome — CHANGES_REQUESTED** [review]
Hi Olivier,

After few small changes I was able to successfully get the quorum device added/modified/deleted in my testing Debian cluster. So definitely good job implementing the module!

I had to also rename the module to have extension `.py` for `ansible-doc` to get it recognized. 
Woudl you please rename the module to `pcs_quorum_qdevice.py`? (you can use command `git mv pcs_quorum_qdevice pcs_quorum_qdevice.py`)

Still before this can be merged some work needs to be done and you can find my comment ans suggestions below and in code line comments.

In addition to those below are the outputs reported by `ansible-test` for code formatting that would be good to address as well.

If you have any questions to my requests feel free to ask. Will be waiting on your response.

Ondrej (온드레이)

~~~
# ansible-test sanity --docker default pcs_quorum_qdevice
...
Running sanity test 'pep8' with Python 3.6
ERROR: Found 5 pep8 issue(s) which need to be resolved:
ERROR: lib/ansible/modules/pcs_quorum_qdevice.py:97:38: E231: missing whitespace after ','
ERROR: lib/ansible/modules/pcs_quorum_qdevice.py:124:8: E111: indentation is not a multiple of four
ERROR: lib/ansible/modules/pcs_quorum_qdevice.py:177:68: E271: multiple spaces after keyword
ERROR: lib/ansible/modules/pcs_quorum_qdevice.py:207:15: E211: whitespace before '['
ERROR: lib/ansible/modules/pcs_quorum_qdevice.py:220:1: E302: expected 2 blank lines, found 1
See documentation for help: https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/pep8.html
~~~
~~~
# ansible-test sanity --docker default pcs_quorum_qdevice
...
Running sanity test 'pylint' with Python 3.6
ERROR: Found 2 pylint issue(s) which need to be resolved:
ERROR: lib/ansible/modules/pcs_quorum_qdevice.py:97:37: bad-whitespace: Exactly one space required after comma             state=dict(required=False,default="present", choices=['present', 'absent']),                                      ^
ERROR: lib/ansible/modules/pcs_quorum_qdevice.py:207:15: bad-whitespace: No space allowed before bracket         result ['msg'] = msg                ^
See documentation for help: https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/pylint.html
~~~

File: library/pcs_quorum_qdevice [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+author: "Olivier Pouilly (@OndrejHome)"
```
[the remark this answers is not in the document]
Comment (OliPou) [reply library/pcs_quorum_qdevice]: Oh yes. Sorry It should be good on the last version.

File: library/pcs_quorum_qdevice [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    description:
+      - "'none' - node list must match existing cluster if cluster should be present"
+      - "'update' - allow qdevice update"
+    default: none
+    required: false
+    choices: ['none', 'update']
+    type: str
+notes:
+   - Tested on Debian 10
+   - "When adding/removing qdevice, make sure to use 'run_once=True' on a cluster node
```
[the remark this answers is not in the document]
Comment (OliPou) [reply library/pcs_quorum_qdevice]: My bad, corrected on the last version

File: library/pcs_quorum_qdevice [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        algo_name = algorithm_name.findall(re_qdevice_defined[0])
+        if len(algo_name) == 0 or algo_name[0] != algorithm:
+            config_qdevice_algo_diff = True
+        else:
+            config_qdevice_algo_diff = False
+
+    update, mismatch_options = False, False
+    msg = ''
+
+    if no_conf:
```
[the remark this answers is not in the document]
Comment (OliPou) [reply library/pcs_quorum_qdevice]: True, Corrected on the last version

File: library/pcs_quorum_qdevice [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        module.fail_json(msg='When creating/updating qdevice you must have a cluster set')
+
+    # if qdevice is not None:
+    #     qdevice_set = qdevice
+    #     qdevice_set_detailed = "host: {} ".format(qdevice)
+
+    # if algorithm is not None:
+    #     algorithm_set = algorithm
+    #     algorithm_set_detailed = "algorithm: {} ".format(algorithm)
+
```
[the remark this answers is not in the document]
Comment (OliPou) [reply library/pcs_quorum_qdevice]: Done :)

File: library/pcs_quorum_qdevice (hunk 1 above)
Comment (OliPou) [reply library/pcs_quorum_qdevice]: Done :)

File: library/pcs_quorum_qdevice (hunk 2 above)
[the remark this answers is not in the document]
Comment (OndrejHome) [reply library/pcs_quorum_qdevice]: looks good, commit 96d1e5d05155157d310665cdd38e5290c3f9f31b, this one is resolved.

File: library/pcs_quorum_qdevice (hunk 3 above)
[the remark this answers is not in the document]
Comment (OndrejHome) [reply library/pcs_quorum_qdevice]: I don't see any commit fixing this, can you please check if it was pushed to repository?

File: library/pcs_quorum_qdevice (hunk 4 above)
[the remark this answers is not in the document]
Comment (OndrejHome) [reply library/pcs_quorum_qdevice]: hmm also here I don't see any commit fixing this, can you please check if it was pushed to repository?

File: library/pcs_quorum_qdevice (hunk 1 above)
Comment (OndrejHome) [reply library/pcs_quorum_qdevice]: also I don't see any changes here. Just to be sure please let me know in which commit it was fixed.

**Comment by OndrejHome** [comment]
Hi @OliPou ,

Thank you for replies and changes. I think that some commits are missing. If you could provide the commit hash to each fix you have done that would be best as then I can check easily which commit is addressing the issue.

File: library/pcs_quorum_qdevice (hunk 3 above)
[the remark this answers is not in the document]
Comment (OliPou) [reply library/pcs_quorum_qdevice]: Sorry Just added commit :
eaffc86

File: library/pcs_quorum_qdevice (hunk 4 above)
[the remark this answers is not in the document]
Comment (OliPou) [reply library/pcs_quorum_qdevice]: Sorry, Just added too:
a35c413

File: library/pcs_quorum_qdevice (hunk 1 above)
Comment (OliPou) [reply library/pcs_quorum_qdevice]: Sorry my bad again:
8896006

File: library/pcs_quorum_qdevice (hunk 4 above)
[the remark this answers is not in the document]
Comment (OndrejHome) [reply library/pcs_quorum_qdevice]: looks ok

File: library/pcs_quorum_qdevice (hunk 1 above)
Comment (OndrejHome) [reply library/pcs_quorum_qdevice]: now it looks better, Thanks!

**Comment by OndrejHome** [comment]
Hi @OliPou ,

Thank you for additional changes. I have once again retested the module on my Debian 10.8/10.9 clusters and it looks good. Before merging I have pushed few cosmetical changes. Code is now merged in `master` and I have added tag `25.0.0` to it.

Thank you for your effort and contribution to this repository!

**Pull request merged**